### PR TITLE
chore: simplify package management

### PR DESCRIPTION
**Pull request by ptab**
## Running in Docker
```
❯ docker run -v $(pwd):/repo -e MPYL_CONFIG_PATH=tests/test_resources/mpyl_config.yml gh-mpyl:local projects lint
✅ tests/projects/dagster-user-code/deployment/project.yml
✅ tests/projects/job/deployment/project.yml
✅ tests/projects/overriden-project/deployment/project-override-first.yml
✅ tests/projects/overriden-project/deployment/project-override-second.yml
✅ tests/projects/overriden-project/deployment/project.yml
✅ tests/projects/sbt-service/deployment/project.yml
✅ tests/projects/service/deployment/project.yml
Validated 7 projects. 7 valid, 0 invalid

Checking for duplicate project names:
  ✅ No duplicate project names found

Checking for missing project ids:
  ✅ All kubernetes projects have a project id

Checking namespace substitution:
  ✅ No wrong namespace substitutions found

Checking whitelisting rules for target PullRequest:
  ✅ No undefined whitelists found

Checking whitelisting rules for target PullRequestBase:
  ✅ No undefined whitelists found

Checking whitelisting rules for target Acceptance:
  ✅ No undefined whitelists found

Checking whitelisting rules for target Production:
  ✅ No undefined whitelists found

Checking for projects depending on themselves:
  ✅ No project with a dependency on itself found
```

## Running with pipenv
```
❯ MPYL_CONFIG_PATH=tests/test_resources/mpyl_config.yml pipenv run cli projects lint
Loading .env environment variables...
✅ tests/projects/dagster-user-code/deployment/project.yml
✅ tests/projects/job/deployment/project.yml
✅ tests/projects/overriden-project/deployment/project-override-first.yml
✅ tests/projects/overriden-project/deployment/project-override-second.yml
✅ tests/projects/overriden-project/deployment/project.yml
✅ tests/projects/sbt-service/deployment/project.yml
✅ tests/projects/service/deployment/project.yml
Validated 7 projects. 7 valid, 0 invalid

Checking for duplicate project names:
  ✅ No duplicate project names found

Checking for missing project ids:
  ✅ All kubernetes projects have a project id

Checking namespace substitution:
  ✅ No wrong namespace substitutions found

Checking whitelisting rules for target PullRequest:
  ✅ No undefined whitelists found

Checking whitelisting rules for target PullRequestBase:
  ✅ No undefined whitelists found

Checking whitelisting rules for target Acceptance:
  ✅ No undefined whitelists found

Checking whitelisting rules for target Production:
  ✅ No undefined whitelists found

Checking for projects depending on themselves:
  ✅ No project with a dependency on itself found
```

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -42,32 +76,17 @@ dill = "==0.3.8"
...
 format-update = "black .  --config pyproject.toml --exclude venv|build/lib"
 test = "pytest -n 4 -W ignore::pytest.PytestCollectionWarning -v tests"
-test-coverage = "bash -c 'coverage run -m pytest -v tests && coverage report -m --sort cover'"
 test-ci = "coverage run -m pytest --junitxml=build/test-result/test.xml"
 test-ci-coverage = "coverage xml -o build/coverage.xml"
-doc = "python -m pdoc -d restructuredtext -o docs/ --favicon https://user-images.githubusercontent.com/1911436/266026788-9c1955a3-3bb4-4372-b8dd-fecc06b8295c.png --logo https://user-images.githubusercontent.com/1911436/266025430-b2cb6249-39e4-4441-b436-5015ad677716.png src/mpyl"
-doc-preview = "python -m pdoc -d restructuredtext --favicon https://user-images.githubusercontent.com/1911436/266026788-9c1955a3-3bb4-4372-b8dd-fecc06b8295c.png --logo https://user-images.githubusercontent.com/1911436/266025430-b2cb6249-39e4-4441-b436-5015ad677716.png src/mpyl"
 check-types = "mypy --explicit-package-bases --check-untyped-defs src/mpyl/"
 check-types-test = "mypy --no-incremental --explicit-package-bases --check-untyped-defs tests/"
-validate = "python validate.py"
```
Comment (Jorg88) [inline Pipfile:63]: I still use this locally before pushing

File: Pipfile [hunk 2]
```diff
@@ -4,36 +4,70 @@ verify_ssl = true
...
+s3transfer = "==0.6.2"
+shellingham = "==1.5.4"
+six = "==1.16.0"
+smmap = "==5.0.1"
+toml = "==0.10.2"
+tqdm = "==4.66.5"
+typer = "==0.12.5"
+typing-extensions = "==4.12.2"
+urllib3 = "==1.26.20"
+websocket-client = "==1.8.0"
```
Comment (Jorg88) [inline Pipfile:60]: Where do all these extra packages come from? We didn't need them before, why now

File: Pipfile [hunk 3]
```diff
@@ -42,32 +76,17 @@ dill = "==0.3.8"
 
 [requires]
 python_version = "3.11"
+#python_full_version = "3.11.8"
 
 [scripts]
-cli = "python run.py"
-cli-ext = "bash -c \"python $(dirname \\$PIPENV_PIPFILE)/src/mpyl/__main__.py $(echo $0) $(echo $@)\""
```
Comment (Jorg88) [inline Pipfile:48]: This is also still needed for running from the monorepo right?

Comment (ptab) [reply Pipfile:48]: I couldn't find [any ocurrences of `cli-ext` in the monorepo](repo:Vandebron/Vandebron cli-ext), so I guess not.

File: Pipfile (hunk 2 above)
Comment (ptab) [reply Pipfile:60]: I regenerated both `Pipfile` and `Pipfile.lock` automatically from `requirements.txt`, and [these packages are defined there](https://github.com/Vandebron/gh-mpyl/blob/7667be9d02d80e3e24599f4f21ac36cb2548bc15/requirements.txt#L52-L55). Why they were added to `requirements.txt` I don't know 🙂

Comment (ptab) [reply Pipfile]: I also just realized that if you created `requirements.txt` from `Pipfile` then there's no point in doing the other way around, so I reverted those changes (and kept only the removal of the unnecessary scripts).

File: Pipfile (hunk 3 above)
Comment (Jorg88) [reply Pipfile:48]: I sometimes still want to quickly run gh-mpyl locally without having to build the image every time, that's what this command is used for. There's no harm in keeping it right?

File: Pipfile (hunk 2 above)
Comment (Jorg88) [reply Pipfile]: It's weird that when you generate pipfile -> requirements -> pipfile you end up with a different pipfile, but whateverrrr

File: Dockerfile [hunk 4]
```diff
@@ -8,13 +8,15 @@ RUN apt-get update && apt-get install -y \
...
-WORKDIR /app/mpyl
-
-COPY requirements.txt requirements.txt
+# USER vdbnonroot  # Enable again after removing git from the src code
 
 # Install the dependencies.
-RUN python -m pip install -r requirements.txt
+RUN pip install pipenv
+COPY Pipfile Pipfile.lock ./
+RUN pipenv install --system --deploy
```
Comment (Jorg88) [inline Dockerfile:16]: how much slower is all of this?

File: README-dev.md [hunk 5]
```diff
@@ -94,7 +94,7 @@ For a shorter feedback loop, you can run the mpyl sourcecode against another rep
 To test the mpyl sourcecode against the peculiarities of your own repository, you can run the following command:
 
 ```shell
-PIPENV_PIPFILE=/<absolute_path_to_mpyl_repo>/Pipfile pipenv run cli-ext build status
+PIPENV_PIPFILE=/<absolute_path_to_mpyl_repo>/Pipfile pipenv run cli build status
```
Comment (Jorg88) [inline README-dev.md:97]: So this is what I meant in the other comment, this doesn't work with the `cli` command. It's what `cli-ext` was made for

File: Pipfile (hunk 3 above)
Comment (ptab) [reply Pipfile:48]: I usually run `pipenv run cli …`, is there a difference?

File: Dockerfile (hunk 4 above)
Comment (ptab) [reply Dockerfile:16]: Not at all:
```
#11 [linux/amd64 3/9] RUN pip install pipenv
#11 DONE 4.8s
#13 [linux/amd64 5/9] RUN pipenv install --system --deploy
#13 DONE 12.2s
```
vs
```
#13 [linux/amd64 5/8] RUN python -m pip install -r requirements.txt
#13 DONE 16.1s
```

Comment (Jorg88) [reply Dockerfile:16]: Nice!

**Review by Jorg88 — APPROVED** [review]

**Pull request merged**